### PR TITLE
ddl, stat: switch to new struct for drop table, add/modify column

### PR DESCRIPTION
**Pull request by fzzf678**
drop table, create/modify column

### What problem does this PR solve?

Issue Number: ref #55723

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/util/schema_change_notifier.go [hunk 1]
```diff
@@ -104,6 +108,55 @@ func (s *SchemaChangeEvent) GetTruncateTableInfo() (
...
+) *SchemaChangeEvent {
+	return &SchemaChangeEvent{
+		tp:           model.ActionDropTable,
+		oldTableInfo: droppedTableInfo,
+	}
+}
+
+// GetDropTableInfo returns the table info of the SchemaChangeEvent whose type is ActionDropTable.
+func (s *SchemaChangeEvent) GetDropTableInfo() (newTableInfo *model.TableInfo) {
+	return s.oldTableInfo
```
Comment (lance6716) [inline pkg/ddl/util/schema_change_notifier.go:123]:
```suggestion
	intest.Assert(s.tp == model.ActionDropTable)
	return s.oldTableInfo
```

File: pkg/ddl/util/schema_change_notifier.go [hunk 2]
```diff
@@ -104,6 +108,55 @@ func (s *SchemaChangeEvent) GetTruncateTableInfo() (
...
+	return &SchemaChangeEvent{
+		tp:           model.ActionAddColumn,
+		newTableInfo: tableInfo,
+		columnInfos:  newColumnInfo,
+	}
+}
+
+// GetAddColumnInfo returns the table info of the SchemaChangeEvent whose type is ActionAddColumn.
+func (s *SchemaChangeEvent) GetAddColumnInfo() (newTableInfo *model.TableInfo, newColumnInfo []*model.ColumnInfo) {
+	return s.newTableInfo, s.columnInfos
```
Comment (lance6716) [inline pkg/ddl/util/schema_change_notifier.go:140]: also add an asert

File: pkg/ddl/util/schema_change_notifier.go [hunk 3]
```diff
@@ -104,6 +108,55 @@ func (s *SchemaChangeEvent) GetTruncateTableInfo() (
...
+	return &SchemaChangeEvent{
+		tp:           model.ActionModifyColumn,
+		newTableInfo: tableInfo,
+		columnInfos:  modifiedColumnInfo,
+	}
+}
+
+// GetModifyColumnInfo returns the table info of and column info the SchemaChangeEvent whose type is ActionModifyColumn.
+func (s *SchemaChangeEvent) GetModifyColumnInfo() (newTableInfo *model.TableInfo, modifiedColumnInfo []*model.ColumnInfo) {
+	return s.newTableInfo, s.columnInfos
```
Comment (lance6716) [inline pkg/ddl/util/schema_change_notifier.go:157]: ditto

**Review by lance6716 — APPROVED** [review]

File: pkg/ddl/util/schema_change_notifier.go (hunk 3 above)
Comment (fzzf678) [reply pkg/ddl/util/schema_change_notifier.go:160]: all fixed

**Review by Rustin170506 — APPROVED** [review]
Looks good to me. Thanks! 👍

**Pull request merged**